src/v/datalake/translation/partition_translator.cc

```diff
@@ -271,8 +271,9 @@ ss::future<> partition_translator::translate_until_stopped() {
               checkpoint_result);
         }
 
-        reset_error = co_await _data_source->update_highest_translated_offset(
-          last_translated_offset, _term, wait_timeout, _as);
+        reset_error
+          = co_await _data_source->replicate_highest_translated_offset(
+            last_translated_offset, _term, wait_timeout, _as);
```

Comment:
For posterity, we discussed offline that maybe this should only update the lag timestamp if we are replicating up to the translation target. I don't think the dust settled on that conversation, but thought I'd mention it here